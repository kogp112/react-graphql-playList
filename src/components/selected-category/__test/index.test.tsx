import * as React from "react";
import * as renderer from "react-test-renderer";
import SelectedCategory from "../index";

let container: JSXElement = null;
type JSXElement = Element | null;

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  container = null;
});

it("should render selected category", () => {
  const onChange = jest.fn();
  const genres = ["Rock", "Punk", "Acidhouse", "Ambientmusic"];

  const component = renderer
  .create(<SelectedCategory genres={genres} handleClick={onChange} />)
  .toJSON();
  expect(component).toMatchSnapshot();
});
