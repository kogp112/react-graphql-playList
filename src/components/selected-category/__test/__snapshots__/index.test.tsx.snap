// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render selected category 1`] = `
<div
  className="MuiGrid-root makeStyles-root-1"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": "blue",
        "color": "white",
      }
    }
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Rock
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": "blue",
        "color": "white",
      }
    }
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Punk
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": "blue",
        "color": "white",
      }
    }
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Acidhouse
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": "blue",
        "color": "white",
      }
    }
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Ambientmusic
    </span>
  </button>
</div>
`;
